### PR TITLE
fix: SDK DeliveryLog.response.response_insertion

### PR DESCRIPTION
**Pull request by danbosnichill**
Previously, the code would do a different transition of the response insertions.  Now it'll reuse the existing response insertions.

I also removed the filling properties from request insertions onto response insertions.  I don't think our customers use that functionality.

TESTING=unit tests

### CODE REVIEW

File: lib/promoted/ruby/client/log_request_builder.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                execution_server: exec_server,
+                server_version: Promoted::Ruby::Client::SERVER_VERSION
+              },
+              request: stripped_request,
+              response: {
+                insertion: @response_insertions
+              }
+            }]
+          end
+          log_req.clean!
```
Comment (danbosnichill) [inline lib/promoted/ruby/client/log_request_builder.rb:52]: This removes empty insertion arrays.  This changed one of the tests.

File: lib/promoted/ruby/client/log_request_builder.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          else
+            @id_generator = IdGenerator.new
+          end
+        end
+
+        def log_request(include_delivery_log:, exec_server:)
+          log_req = {
+            platform_id: @request[:platform_id],
+            user_info: @request[:user_info],
+            # TODO - what timing should be used on this?
```
Comment (danbosnichill) [inline lib/promoted/ruby/client/log_request_builder.rb:20]: Not sure the small difference matters.

File: spec/promoted/ruby/client/pager_spec.rb [hunk 3]
```diff
@@ -145,21 +145,6 @@
       expect(res[2][:position]).to eq 2
     end
 
-    it "does not touch positions that are already set" do
-      @insertions[0][:position] = 100
```
Comment (danbosnichill) [inline spec/promoted/ruby/client/pager_spec.rb:149]: Request insertion shouldn't set position.

**Review by prm-james-hill — APPROVED** [review]
Nothing looks obviously wrong, but I don't know Ruby. Guessing you should wait for Scott in any event

**Pull request merged**